### DISCUSSION TITLE
greater control for mouse acceleration

### DISCUSSION THREAD

**Discussion by khlsvr** [AntiMicroX/antimicrox#720] [Ideas]
TLDR; Is it possible to flick the gamepad stick (max distance 32000) and have mouse cursor move only very short distance, while if held longer it would gain a way faster speed?

I have this cheap gamepad SHANWAN. The control of the sticks is not sensitive, meaning that it is difficult to move the axis only a certain amount. For this reason I often end up moving axis the max amount (32000). Now for controlling the mouse on a computer desktop, this is problematic. If I want to move a very short distance, it will easily go too far with reasonable speed settings. Then if I set very slow speed settings, it's too slow for moving a longer distance.

I also use the linux feature of "mousekeys" to simulate mouse with numpad. I use the tool xkbset to set the mouse speed. With this tool, I can have a better mouse control with basically on/off keyboard keys than I can with antimicrox and gamepad sticks. I run mousekeys on numpad with the following values in xkbset: `xkbset ma 80 2 200 5 2000`. So with this setting, when I press numpad key `8` down for example, it will go slowly the first 0.2 seconds or such and then quite swiftly accelerates. This means that when I do rather short press downs of the keys, I can move short distances, because the speed is still slow, but if I press down the button for a longer period, it will gain good speed for longer distance.

The above speed - acceleration formula with antimicrox would solve this non-sensitive gamepad stick issue also. I have tried to tinker with all the settings, and have read https://github.com/AntiMicroX/antimicrox//wiki/Mouse-Settings but I just don't get what I should do with those settings to achieve this. With that delta acceleration thing I seem to be able to control the speed depending how quickly I turn the stick to certain direction, but this is not helping. The sticks are so unsensitive that I don't get a control of the speed with this. I need a setting that however fast I floor the stick to some direction, the first x seconds it would go slowly and only then accelerate to a higher speed.

Does this require a new feature in the settings, or can I somehow achieve it with current available settings?
